eslint.config.mjs

```diff
@@ -0,0 +1,135 @@
+import importPlugin from 'eslint-plugin-import-x';
+import react from 'eslint-plugin-react';
+import globals from 'globals';
+import babelParser from '@babel/eslint-parser';
+import eslint from '@eslint/js';
+
+export default [
+  {
+    ignores: [
+      '**/.vscode/',
+      '.github/',
+      'media/',
+      'static/',
+      '**/coverage/',
+      'docs/_build/',
+      '**/package-lock.json',
+      '**/specs/',
+      '**/?venv/',
+      'translate/dist/',
+      // Jinja templates
+      'translate/public/translate.html',
+      '**/templates/**/*.html',
+      // Vendored code
+      'error_pages/css/blockrain.css',
+      'error_pages/js/',
+      'pontoon/base/static/css/boilerplate.css',
+      'pontoon/base/static/css/fontawesome-all.css',
+      'pontoon/base/static/css/nprogress.css',
+      'pontoon/base/static/js/lib/',
+    ],
+  },
+  eslint.configs.recommended,
+  react.configs.flat.recommended,
+  {
+    plugins: {
+      react,
+      import: importPlugin,
+    },
+
+    languageOptions: {
+      globals: {
+        ...globals.browser,
+        ...globals.jest,
+        gettext: false,
+        ngettext: false,
+        interpolate: false,
+        l: false,
+        expect: false,
+        test: false,
+        browser: false,
+        jest: false,
+        Promise: false,
+        Set: false,
+        URLSearchParameters: false,
+        FormData: false,
+        require: false,
+        shortcut: false,
+        sorttable: false,
+        $: false,
+        Pontoon: false,
+        jQuery: false,
+        Clipboard: false,
+        Chart: false,
+        NProgress: false,
+        diff_match_patch: false,
+        Highcharts: false,
+        Sideshow: false,
+        editor: false,
+        DIFF_INSERT: false,
+        DIFF_EQUAL: false,
+        DIFF_DELETE: false,
+        ga: false,
+        process: false,
+        generalShortcutsHandler: true,
+        traversalShortcutsHandler: true,
+        editorShortcutsHandler: true,
+        showdown: true,
+      },
+
+      parser: babelParser,
```

Comment:
Do we really still need Babel to parse the JS sources? That seems unlikely, esp. as it's not required for JSX.

Reply:
That I don't really know. If I use `tsParser` like in the main config it works, if I drop it I get 22 errors of these types

```
error  Parsing error: Unexpected token {
error  Parsing error: Unexpected token ..
error  Parsing error: Unexpected token ?
```

Reply:
When testing this, those go away when we also drop the unnecessary `ecmaVersion`. I'll add a suggestion below for how to simplify this that seems to work for me.